### ISSUE TITLE
BufferRelatedType shouldn't be read as identifiers

### ISSUE THREAD

**Issue by saschanaz** [w3c/webidl2.js#251]
https://heycam.github.io/webidl/#index-prod-BufferRelatedType

```
BufferRelatedType ::
    ArrayBuffer
    DataView
    Int8Array
    Int16Array
    Int32Array
    Uint8Array
    Uint16Array
    Uint32Array
    Uint8ClampedArray
    Float32Array
    Float64Array
```

These are keywords so the following is illegal:

```
interface Int32Array {};
```

**Comment by saschanaz** [comment]
BufferRelatedType is not exclusively used anywhere, though, whereas StringType is.

**Comment by saschanaz** [comment]
Also closed by #439
